javascript/extractor/src/com/semmle/js/extractor/ASTExtractor.java

```diff
@@ -786,11 +838,31 @@ public Label visit(AssignmentExpression nd, Context c) {
       return key;
     }
 
+    // set to determine which BinaryExpression has been extracted as regexp
+    private Set<Expression> extractedAsRegexp = new HashSet<>();
+
     @Override
     public Label visit(BinaryExpression nd, Context c) {
       Label key = super.visit(nd, c);
+      extractedAsRegexp.add(nd.getLeft());
+      extractedAsRegexp.add(nd.getRight());
       visit(nd.getLeft(), key, 0);
       visit(nd.getRight(), key, 1);
+      if (extractedAsRegexp.contains(nd)) {
```

Comment:
Could you factor the RegExp-extraction part into an appropriately named method and call into that when it should be extracted as a regexp? This code applies to all `BinaryExpression` instances and this bailout-style looks a bit out of place here.